tests/projects/projects_namespace_test.py

```diff
@@ -1,16 +1,201 @@
+from faker import Faker
 from flask import json
 from flask.testing import FlaskClient
 from pytest_mock import MockFixture
 
+from time_tracker_api.projects.projects_model import PROJECT_TYPE
 
-def test_list_all_elements(client: FlaskClient, mocker: MockFixture):
+fake = Faker()
+
+valid_project_data = {
+    "name": fake.company(),
+    "description": fake.paragraph(),
+    "type": fake.word(PROJECT_TYPE.valid_type_values()),
+}
+fake_project = ({
+    "id": fake.random_int(1, 9999)
+}).update(valid_project_data)
+
+
+def test_create_project_should_succeed_with_valid_request(client: FlaskClient, mocker: MockFixture):
+    from time_tracker_api.projects.projects_namespace import project_dao
+    repository_create_mock = mocker.patch.object(project_dao.repository,
+                                                 'create',
+                                                 return_value=fake_project)
+
+    response = client.post("/projects", json=valid_project_data, follow_redirects=True)
+
+    assert 201 == response.status_code
+    repository_create_mock.assert_called_once_with(valid_project_data)
+
+
+def test_create_project_should_reject_bad_request(client: FlaskClient, mocker: MockFixture):
     from time_tracker_api.projects.projects_namespace import project_dao
-    repository_find_all_mock = mocker.patch.object(project_dao.repository, 'find_all', return_value=[])
+    invalid_project_data = valid_project_data.copy().update({
+        "type": 'anything',
+    })
+    repository_create_mock = mocker.patch.object(project_dao.repository,
+                                                 'create',
+                                                 return_value=fake_project)
+
+    response = client.post("/projects", json=invalid_project_data, follow_redirects=True)
+
+    assert 400 == response.status_code
+    repository_create_mock.assert_not_called()
+
+
+def test_list_all_projects(client: FlaskClient, mocker: MockFixture):
```

Comment:
What about `test_get_all_projects` or `test_get_projects`?

Reply:
The description in the comments says list. As long as it makes sense and it is not ambiguous we are Ok.